### PR TITLE
sync_point (ticdc): add new config and change sync point table struct

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: ref #6977 

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sink/mysql/mysql_syncpoint_store.go [hunk 1]
```diff
@@ -198,21 +216,35 @@ func (s *mysqlSyncpointStore) SinkSyncpoint(ctx context.Context,
...
 		return cerror.WrapError(cerror.ErrMySQLTxnError, err)
 	}
+
+	query = fmt.Sprintf(
+		"DELETE IGNORE FROM "+schemaName+"."+syncPointTableName+"WHERE cf like '%s' and creat_at < (NOW() - INTERVAL %.2f SECOND)",
+		id.Namespace+"_"+id.ID,
+		s.syncPointRetention.Seconds())
+	_, err = tx.Exec(query)
+	if err != nil {
+		err2 := tx.Rollback()
```
Comment (nongfushanquan) [inline cdc/sink/mysql/mysql_syncpoint_store.go:236]: if failed to remove the old sync-point recorder , writing a log and ignoring this error  is ok

File: cdc/owner/changefeed.go [hunk 2]
```diff
@@ -737,7 +737,7 @@ func (c *changefeed) handleBarrier(ctx cdcContext.Context) (uint64, error) {
 		if !blocked {
 			return barrierTs, nil
 		}
-		nextSyncPointTs := oracle.GoTimeToTS(oracle.GetTimeFromTS(barrierTs).Add(c.state.Info.SyncPointInterval))
+		nextSyncPointTs := oracle.GoTimeToTS(oracle.GetTimeFromTS(barrierTs).Add(c.state.Info.Config.SyncPointInterval))
```
Comment (nongfushanquan) [inline cdc/owner/changefeed.go:740]: the min value of SyncPointInterval  should  be set

File: cdc/sink/mysql/mysql_syncpoint_store.go (hunk 1 above)
Comment (asddongmen) [reply cdc/sink/mysql/mysql_syncpoint_store.go]: got it

File: cdc/owner/changefeed.go (hunk 2 above)
Comment (asddongmen) [reply cdc/owner/changefeed.go:740]: done

File: pkg/config/replica_config.go [hunk 3]
```diff
@@ -120,12 +146,31 @@ func (c *replicaConfig) fillFromV1(v1 *outdated.ReplicaConfigV1) {
...
 		if err != nil {
 			return err
 		}
 	}
+	// check sync point config
+	if c.EnableSyncPoint {
+		if c.SyncPointInterval < minSyncPointInterval {
+			return cerror.ErrInvalidReplicaConfig.
+				FastGenByArgs(
+					fmt.Sprintf("The SyncPointInterval:%s must large than %s",
```
Comment (zhaoxinyu) [inline pkg/config/replica_config.go:161]:
```suggestion
					fmt.Sprintf("The SyncPointInterval:%s must be larger than %s",
```

File: pkg/config/replica_config.go [hunk 4]
```diff
@@ -120,12 +146,31 @@ func (c *replicaConfig) fillFromV1(v1 *outdated.ReplicaConfigV1) {
...
+						c.SyncPointInterval.String(),
+						minSyncPointInterval.String()))
+		}
+		if c.SyncPointRetention < minSyncPointRetention {
+			return cerror.ErrInvalidReplicaConfig.
+				FastGenByArgs(
+					fmt.Sprintf("The SyncPointRetention:%s must large than %s",
```
Comment (zhaoxinyu) [inline pkg/config/replica_config.go:168]:
```suggestion
					fmt.Sprintf("The SyncPointRetention:%s must be larger than %s",
```

File: tests/integration_tests/syncpoint/conf/changefeed.toml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+enable-sync-point = true
+sync-point-interval = "30s"
```
Comment (zhaoxinyu) [inline tests/integration_tests/syncpoint/conf/changefeed.toml:2]: How is the "30s" converted to time.Duration? Can the user write this as '30sec' or '30 sec' or '30 s'?

Comment (asddongmen) [reply tests/integration_tests/syncpoint/conf/changefeed.toml:2]: no, the doc will tell the users they can only use format like: "3h4m30s"

Comment (zhaoxinyu) [reply tests/integration_tests/syncpoint/conf/changefeed.toml:2]: We can keep this configuration item similar as `consistent.flush-interval` (it is in millisecond) and write it as follows:

```
sync-point-interval = 30
```

**Comment by zhaoxinyu** [comment]
Rest LGTM

**Review by zhaoxinyu — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**